### PR TITLE
fyodorov example tests

### PR DESCRIPTION
**Pull request by youRaFyodorov**

### CODE REVIEW

**Review by LiudmilaZhdanok — CHANGES_REQUESTED** [review]
watch the new video in Alumni from the stand-up on how to delete workspace.xml

**Review by olgaz-skiba — APPROVED** [review]

File: homework/java/fyodorov/lecture2/task1/pom.xml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+                        <id>jacoco-check</id>
+                        <goals>
+                            <goal>check</goal>
+                        </goals>
+                        <configuration>
+                            <rules>
+                                <rule>
+                                    <limits>
+                                        <limit>
+                                            <minimum>0.0</minimum>
```
Comment (integratorBY) [inline homework/java/fyodorov/lecture2/task1/pom.xml:65]: set 0.6

File: homework/java/fyodorov/lecture2/task1/src/main/java/by/jrr/mytest/bean/Dog.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
+package by.jrr.mytest.bean;
```
Comment (integratorBY) [inline homework/java/fyodorov/lecture2/task1/src/main/java/by/jrr/mytest/bean/Dog.java:1]: change package name

File: homework/java/fyodorov/lecture2/task1/src/main/java/by/jrr/mytest/bean/Dog.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+public class Dog {
+
+    int age;
```
Comment (integratorBY) [inline homework/java/fyodorov/lecture2/task1/src/main/java/by/jrr/mytest/bean/Dog.java:5]: use private

File: homework/java/fyodorov/lecture2/task1/src/main/java/by/jrr/mytest/service/DogService.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
+package by.jrr.mytest.service;
+import by.jrr.mytest.bean.Dog;
+
+public class DogService {
+
+    Dog dog;
```
Comment (integratorBY) [inline homework/java/fyodorov/lecture2/task1/src/main/java/by/jrr/mytest/service/DogService.java:6]: use private

File: homework/java/fyodorov/lecture2/task1/src/main/java/by/jrr/mytest/service/DogService.java [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+    public DogService(Dog dog) {
+        this.dog = dog;
+    }
+
+    public String bark (){
+        return dog.getName () + " BARK!";
+    }
```
Comment (integratorBY) [inline homework/java/fyodorov/lecture2/task1/src/main/java/by/jrr/mytest/service/DogService.java:14]: use method .conact()

File: homework/java/fyodorov/lecture2/task1/src/main/java/by/jrr/mytest/service/DogService.java [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+    public double convertPriceToBelarus (double exchangeRate){
+        double convertedPrice = dog.getPrice()*exchangeRate;
+        if (convertedPrice > 1000){
+            convertedPrice = 0.97 * convertedPrice;
+        }
+        return convertedPrice;
```
Comment (integratorBY) [inline homework/java/fyodorov/lecture2/task1/src/main/java/by/jrr/mytest/service/DogService.java:20]: avoid to use magic string

File: homework/java/fyodorov/lecture2/task1/src/main/java/by/jrr/mytest/service/DogServiceUtil.java [hunk 7]
```diff
@@ -0,0 +1,8 @@
+package by.jrr.mytest.service;
+import by.jrr.mytest.bean.Dog;
+
+public class DogServiceUtil {
+    public static String bark (Dog dog){
+        return dog.getName () + " BARK!";
```
Comment (integratorBY) [inline homework/java/fyodorov/lecture2/task1/src/main/java/by/jrr/mytest/service/DogServiceUtil.java:6]: magic string . psfs - short key

Comment (integratorBY) [inline homework/java/fyodorov/lecture2/task1/src/main/java/by/jrr/mytest/service/DogServiceUtil.java:6]: create class with name "DogLanguage" and set magic string into this

File: homework/java/fyodorov/lecture2/task1/src/test/java/by/jrr/mytest/service/DogServiceTest.java [hunk 8]
```diff
@@ -0,0 +1,45 @@
+package by.jrr.mytest.service;
+
+import by.jrr.mytest.bean.Dog;
+import org.junit.Assert;
+import org.junit.Test;
+
+import static org.junit.Assert.*;
```
Comment (integratorBY) [inline homework/java/fyodorov/lecture2/task1/src/test/java/by/jrr/mytest/service/DogServiceTest.java:7]: import only static method

Comment (integratorBY) [reply homework/java/fyodorov/lecture2/task1/src/test/java/by/jrr/mytest/service/DogServiceTest.java:7]: dont use *

File: homework/java/fyodorov/lecture2/task1/src/test/java/by/jrr/mytest/service/DogServiceTest.java [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+
+public class DogServiceTest {
+
+    @Test
+    public void bark() {
+        Dog dog = new Dog ();
```
Comment (integratorBY) [inline homework/java/fyodorov/lecture2/task1/src/test/java/by/jrr/mytest/service/DogServiceTest.java:13]: use var

File: homework/java/fyodorov/lecture2/task1/src/test/java/by/jrr/mytest/service/DogServiceTest.java [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+        dog.setName("Volodya");
+        String expected = "Volodya BARK!";
+
+        DogService dogService = new DogService(dog);
```
Comment (integratorBY) [inline homework/java/fyodorov/lecture2/task1/src/test/java/by/jrr/mytest/service/DogServiceTest.java:17]: use var

File: homework/java/fyodorov/lecture2/task1/src/test/java/by/jrr/mytest/service/DogServiceTest.java [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+        String actual = dogService.bark();
+        Assert.assertEquals("dogServiceTest", expected, actual);
+    }
+
+    @Test
+    public void convertPriceToBelarus_noDiscont() {
+        Dog dog = new Dog ();
+        dog.setPrice (300);
+        double exchangeRate = 2.54;
+        double expected = 762; //762
```
Comment (integratorBY) [inline homework/java/fyodorov/lecture2/task1/src/test/java/by/jrr/mytest/service/DogServiceTest.java:27]: remove comment

File: homework/java/fyodorov/lecture2/task1/src/test/java/by/jrr/mytest/service/DogServiceTest.java [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+        String expected = "Volodya BARK!";
+
+        DogService dogService = new DogService(dog);
+        String actual = dogService.bark();
+        Assert.assertEquals("dogServiceTest", expected, actual);
+    }
+
+    @Test
+    public void convertPriceToBelarus_noDiscont() {
+        Dog dog = new Dog ();
```
Comment (integratorBY) [inline homework/java/fyodorov/lecture2/task1/src/test/java/by/jrr/mytest/service/DogServiceTest.java:24]: create private method to create `var dog = new Dog()`

File: homework/java/fyodorov/lecture2/task1/src/test/java/by/jrr/mytest/service/DogServiceTest.java [hunk 13]
```diff
@@ -0,0 +1,45 @@
...
+        DogService dogService = new DogService(dog);
+        double actual = dogService.convertPriceToBelarus(exchangeRate);
+        Assert.assertEquals(expected, actual, 0.01);
+    }
+    @Test
+    public void convertPriceToBelarus_shouldReturnWithDiscont() {
+        Dog dog = new Dog ();
+        dog.setPrice (1200);
+        double exchangeRate = 2.54;
+        double expected = 2956.56; //3048
```
Comment (integratorBY) [inline homework/java/fyodorov/lecture2/task1/src/test/java/by/jrr/mytest/service/DogServiceTest.java:38]: delete comment

**Review by integratorBY — APPROVED** [review]
lgfm. but change some code

File: homework/java/fyodorov/lecture2/task1/src/main/java/by/jrr/mytest/service/DogService.java [hunk 14]
```diff
@@ -0,0 +1,22 @@
...
+public class DogService {
+
+    Dog dog;
+
+    public DogService(Dog dog) {
+        this.dog = dog;
+    }
+
+    public String bark (){
+        return dog.getName () + " BARK!";
```
Comment (javaGuruBY) [inline homework/java/fyodorov/lecture2/task1/src/main/java/by/jrr/mytest/service/DogService.java:13]: You can create class DogLanguage with static fields like this:
```
public class DogLanguage() {
    public static final String BARKING = "BARK!";
}
```

Than in code you can address like this:
```
public String bark() {
        return String.format("%s %s", dog.getName(), DogLanguage.BARKING);
}
```

**Review by javaGuruBY — APPROVED** [review]
minor comments

**Review by LiudmilaZhdanok — APPROVED** [review]
lgfm

**Pull request merged**